src/lib/fetchChatter.ts

```diff
@@ -41,22 +41,46 @@ export class FetchChatter {
         return this.fetcher.fetchJSON(fetchUrl, fetchOptions);
     }
 
-    private confirmCommunityID(){
-        if(!this.options.sfdcCommunityID){
-            throw 'SFDC Community ID is required to fetch Chatter';
-        }
+    create(resource:string, body:any): Promise<any> {
+        this.confirmCommunityID();
+        let fetchUrl = urlJoin(this.getBaseChatterURL(), resource);
+
+        let bodyJSON = JSON.stringify(body);
+        let fetchOptions = {
+            headers: { 'Content-Type': 'application/json' },
+            method: 'POST',
+            body: bodyJSON
+        };
+        return this.fetcher.fetchJSON(fetchUrl, fetchOptions);
     }
 
-    post(post: any): Promise<any> {
+    update(resource:string, body:any): Promise<any> {
         this.confirmCommunityID();
-        let fetchUrl = urlJoin(this.getBaseChatterURL(), 'feed-elements');
-        
-        let bodyJSON = JSON.stringify(post);
+        let fetchUrl = urlJoin(this.getBaseChatterURL(), resource);
+
+        let bodyJSON = JSON.stringify(body);
         let fetchOptions = {
             headers: { 'Content-Type': 'application/json' },
             method: 'POST',
             body: bodyJSON
         };
         return this.fetcher.fetchJSON(fetchUrl, fetchOptions);
     }
+
+    delete(resource:string): Promise<any> {
+        this.confirmCommunityID();
+        let fetchUrl = urlJoin(this.getBaseChatterURL(), resource);
+
+        let fetchOptions = {
+            headers: { 'Content-Type': 'application/json' },
+            method: 'DELETE'
+        };
+        return this.fetcher.fetchJSON(fetchUrl, fetchOptions);
+    }
+    
+    private confirmCommunityID(){
```

Comment:
privateCommunityID should be up as close as possible to where it is called first (line 34)

This is based off of Clean Code's The Stepdown Rule:

> We want the code to read like a top-down narrative. We want every function to be followed by those at the next level of abstraction so that we can read the program, descending one level of abstraction at a time as we read down the list of functions. I call this The Step-down Rule.